### PR TITLE
 Support vertex color transfer

### PR DESCRIPTION
**Pull request by hooyuser**

### CODE REVIEW

**Comment by hooyuser** [comment]
Here is a test file
create_vertex_color

[zeno_vdb_slice.blend.txt](https://github.com/zenustech/zenoblend/files/7105757/zeno_vdb_slice.blend.txt)

File: zenoblend/main.cpp [hunk 1]
```diff
@@ -292,25 +292,65 @@ PYBIND11_MODULE(pylib_zenoblend, m) {
...
         auto mesh = reinterpret_cast<zeno::BlenderMesh *>(meshPtr);
         auto loop = reinterpret_cast<MLoop *>(loopPtr);
+
         for (int i = 0; i < loopCount; i++) {
             loop[i].v = mesh->loop[i];
             loop[i].e = 0;
         }
+
+        if(loopColorPtr) {
+            auto iter = mesh->vert.attrs.find("clr");
```
Comment (hooyuser) [inline zenoblend/main.cpp:318]: "clr" can be either `float3` or `float`, which then will be converted into blender mesh vertex color.

**Review by archibate — APPROVED** [review]
Thanks! I adjust the internal logic a little bit: vertex colors are considered to be loop attribute, and multiple vertex colors are supported.

**Comment by hooyuser** [comment]
> Thanks! I adjust the internal logic a little bit: vertex colors are considered to be loop attribute, and multiple vertex colors are supported.

Nice work! I wonder if we have some nodes like "Attribute Convert Node" in Blender Geometry Nodes, which can convert a vertex attribute into a loop attribute. Since some zeno nodes will produce `clr` attribute as a vertex attribute, this can enable users to bring the `clr` attribute back to blender mesh.

**Comment by hooyuser** [comment]
If we are going to convert a vertex attribute `clr` into a loop attribute `Col`, it will look like this.

attribute convert

**Comment by archibate** [comment]
> `clr` attribute as a vertex attribute, this kind of node can enable users to bring the `clr` vertex attribute to blender mesh vertex color.

FYI we currently automatically do this `vertex-to-loop` transfer in `PrimitiveToBMesh` when `has_vert_color` is on.

**Comment by archibate** [comment]
Oh I see, so the reason to have separate `loop` attributes is for face that sharing same vertex does not have to share the same, e.g., vertex normal, allowing flat/smooth shading easily.

**Comment by hooyuser** [comment]
@archibate I find when `has_vert_color` is on, all the zeno point attribute will be converted into bmesh vertex color. So we don't have any point attribute for bmesh in that case. In blender, vertex colors are actually special face corner attributes. I think we could only cast the point attribute named `clr` into bmesh vertex color. 

Also I find gamma correction is still necessary? I will give a screenshot to deal with this then.

**Comment by hooyuser** [comment]
As the following screenshot shows, `@clr = 0.5` will be transformed in to `clr = 0.212`. When 8-bit color of the `char` format is converted into color of the `float` format, Blender will make gamma correction. It appears Blender always regards color of the `char` format as non-linear encoded color.

gamma

**Pull request merged**